File: linked_list.c
#include<string.h>
#include<stdio.h>
#include<stdlib.h>
#include<boolean.h>
struct node{
int data;
int key;
struct node *next;
}
struct node *head=NULL;
struct node *current=NULL;
void printlist()
{
printf("[");
struct node *ptr=head;
while(*ptr!=NULL){
printf("%d,%d",ptr->key,ptr->data);
ptr=ptr->next;
}
printf("]");
} 
void inserfirst(int key,int data)
{
struct node *link= (struct node *)malloc(sizeof(struct node));
link->key=key;
link->data=data;
link->next=head;
head=link;
}
struct node* deletefirst()
{
struct node* templink=head;
head=head->next;
return templink;
}
bool isempty()
return head==NULL;
int length()
{
int length=0;
struct node *current;
for(current=head;current!=NULL;current=current->next)
length++;
return length;
}
struct node * find(int key)
{
struct node * current=head;
if(head==NULL)
{
return NULL;
}
while(current->key!=key)
{
if(current->next==NULL)
return NULL;
else
current=current->next;
}
return current;
}
struct node* delete(int key)
{
struct node* current=head;
struct node* previous=NULL;
if(head==NULL)
return NULL;
while(current->key!=key)
{
if(current->next==null)
{
return NULL;
else
previous=current;
current=current->next;
}
}
if(current==head)
head=head->next;
else
previous->next=current->next
return current;
}
main()
{
insertfirst(1,10);
insertfirst(2,20);
insertfirst(3,30);
insertfirst(4,40);
printf("original list:");
printlist();
while(!=isempty())
{
struct node *temp=deletefirst();
printf(deleted value:);
printf("(%d,%d)",temp->key,temp->data);
}
printf("after deleteing all items,list:");
printlist();
insertfirst(1,10);
insertfirst(2,20);
insertfirst(3,30);
insertfirst(4,40);
insertfirst(5,50);
printf("Restored list:)';
printlist();
struct node* foundlink=find(3);
if(foundlink!=NULL){
printf("element found");
printf("(%d,%d)",foundlink->link,foundlink->data);
}
else
printf("Element not found");
delete(4);
printf("after deleting an element");
printlist();
}

File: linkedlist.js
// class Node{
// 	constructor(element){
// 		this.element = element;
// 		this.next = null;
// 	}
// }

// class LinkedList{
// 	contructor(){
// 		this.head = null;
// 		this.size = 0;
// 	}
// 	add(element){
// 		var node = new Node(element);
// 		var current;
// 		if(this.head == null){
// 			this.head = node;
// 		} else {
// 			current = this.head;
// 			while(current.next){
// 				current = current.next;
// 			}
// 			current.next = node;
// 		}
// 		this.size++;
// 	}
// 	printList(){
// 		var curr = this.head;
// 		while(curr){
// 			console.log(curr.element);
// 			curr = curr.next;
// 		}
// 	}
// 	insertAt(element,index){
// 		if(index >0&& index > this.size){
// 			return false;
// 		} else {
// 			var n = new Node(element);
// 			if(index == 0){
// 				n.next = this.head;
// 				this.head = n;
// 			} else {
// 				var curr, prev;
// 				curr = this.head;
// 				var it = 0;
// 				while (it < index){
// 					prev = curr;
// 					curr = curr.next;
// 					it++;
// 				}
// 				n.next = curr;
// 				prev.next = n;
// 			}
// 			this.size++;
// 		}
// 	}
// 	findCycle(l){
// 		var slow = this.head,fast = this.head;
// 		while(fast && fast.next){
// 			fast = fast.next.next;
// 			slow = slow.next;
// 			if(slow== fast){
// 				return true
// 			} 
// 				return false;
// 		}
// 	}

// }

// var ll = new LinkedList();
// ll.add(10)
// ll.add(30)
// ll.add(40)
// ll.add(50)
// ll.printList();
// ll.insertAt(20,1)
// ll.printList();
// var x= ll.findCycle(ll)
// console.log(x);


var obj={}
var n= 3
for(var i=0;i<n;i++){
	obj[i] = [i,i]
}
console.log(obj)
